### PR TITLE
feat: send presence to extension on compile

### PR DESCRIPTION
**Pull request by veryCrunchy**

### CODE REVIEW

File: packages/cli/src/util/socket.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    this.ws = new WebSocketServer({ port: 3613 });
+
+    this.ws.on("connection", (ws) => {
+      console.log(prefix, chalk.greenBright("connected to extension"));
+      ws.on("error", console.error);
+      ws.on("message", (m) => {
+        if (m.toString() === "disconnecting")
+          console.log(prefix, chalk.redBright("extension disconnected"));
+        if (m.toString() === "received")
+          console.log(prefix, chalk.greenBright.dim("presence updated in extension"));
```
Comment (Timeraa) [inline packages/cli/src/util/socket.ts]: Shouldn't the start of a String be Uppercase?

**Review by Timeraa — CHANGES_REQUESTED** [review]

File: packages/cli/src/util/socket.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import chalk from "chalk";
+import { prefix } from "./prefix.js";
+import { WebSocket, WebSocketServer } from "ws";
+
+class WebSocketManager {
+  private static instance: WebSocketManager;
+  private ws: WebSocketServer | null = null;
+
+  private constructor() {
+    this.ws = new WebSocketServer({ port: 3613 });
```
Comment (Bas950) [inline packages/cli/src/util/socket.ts]: Change port to 3021
```suggestion
    this.ws = new WebSocketServer({ port: 3021 });
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: packages/cli/src/util/socket.ts (hunk 1 above)
Comment (veryCrunchy) [reply packages/cli/src/util/socket.ts]: Sure

File: packages/cli/src/actions/modify.ts [hunk 3]
```diff
@@ -199,9 +201,34 @@ class Compiler {
         console.error(error.message);
       }
 
-      if (compilation.errors.length === 0)
-        return console.log(prefix, chalk.greenBright("Successfully compiled!"));
-      else
+      if (compilation.errors.length === 0) {
+        console.log(prefix, chalk.greenBright("Successfully compiled!"));
+        const path = presencePath + "/dist";
+        let files = readdirSync(path);
```
Comment (Bas950) [inline packages/cli/src/actions/modify.ts]: let? I dont see it getting re-assigned

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**